test/csv/parse/test_convert.rb

```diff
@@ -107,4 +107,69 @@ def test_empty_value
     assert_equal([nil, "empty", "a"],
                  CSV.parse_line(',"",a', empty_value: "empty"))
   end
+
+  sub_test_case("#quoted?") do
+    def setup
+      @preserving_converter = lambda do |field, info|
+        f = field.encode(CSV::ConverterEncoding)
+        return f if info.quoted?
+        begin
+          Integer(f, 10)
+        rescue
+          f
+        end
+      end
+
+      @quoted_header_converter = lambda do |field, info|
+        f = field.encode(CSV::ConverterEncoding)
+        return f if info.quoted?
+        f.to_sym
+      end
+    end
+
+    def test_parse_line
+      row = CSV.parse_line('1,"2",3', converters: @preserving_converter)
+      assert_equal([1, "2", 3], row)
+    end
+
+    def test_parse
+      expected = [["quoted", "unquoted"], ["109", 1], ["10A", 2]]
+      rows = CSV.parse(<<~CSV, converters: @preserving_converter)
+        "quoted",unquoted
+        "109",1
+        "10A",2
+      CSV
+      assert_equal(expected, rows)
+    end
+
+    def test_alternating_quote
+      row = CSV.parse_line('"1",2,"3"', converters: @preserving_converter)
+      assert_equal(['1', 2, '3'], row)
+    end
+
+    def test_parse_headers
+      expected = [["quoted", :unquoted], ["109", "1"], ["10A", "2"]]
+      table = CSV.parse(<<~CSV, headers: true, header_converters: @quoted_header_converter)
+        "quoted",unquoted
+        "109",1
+        "10A",2
+      CSV
+      assert_equal(expected, table.to_a)
+    end
+
+    def test_parse_with_string_headers
+      expected = [["quoted", :unquoted], %w[109 1], %w[10A 2]]
+      table = CSV.parse(<<~CSV, headers: '"quoted",unquoted', header_converters: @quoted_header_converter)
+        "109",1
+        "10A",2
+      CSV
+      assert_equal(expected, table.to_a)
+    end
+  end
+
+  sub_test_case("converter with #encode for Symbols") do
+    def test_encode_symbol
+      assert_nothing_raised { CSV.parse("1,2", headers: ["string", :symbol] , header_converters: :symbol) }
```

Comment:
It seems that this is not related to this pull request.
Could you remove this from this pull request? If this is needed, please open another pull request.

Reply:
Thank you. I removed last one and create modified commit.

I will consider more for another pull request about converting when specified header has a Symbol.